Lib/test/test_exceptions.py

```diff
@@ -1170,6 +1170,12 @@ def inner():
             self.fail("RecursionError not raised")
         self.assertEqual(wr(), None)
 
+    def test_errno_ENOTEMPTY(self):
+        with self.assertRaises(OSError) as cm:
```

Comment:
Wait, why not expecting DirectoryNotEmptyError?

Reply:
I tried but I can't assert DirectoryNotEmptyError here. Am I missing something? I will pick it later as a subclass of `OSError`.

Reply:
If the test fails, it sounds like there is a bug in your PR. Moreover, "with self.assertRaises(OSError) as cm:" should only surround "os.rmdir(dirname)".

Reply:
I will investigate.